### PR TITLE
Add the initial WebServerControl

### PR DESCRIPTION
**Pull request by friedmud**
Initial implementation of the `WebServerControl`

See the docs for the object (once it builds) for a full explanation.  What it does is start a webserver that exposes a REST API so that any external application/script can Control the execution of MOOSE.

Right now there are four things you can do:
- See if MOOSE is waiting for external input
- Set the value of a controlled variable
- Get the value of any Postprocessor
- Tell MOOSE to continue solving

This is only the beginning of this capability - look for more to come as people need it

refs #23359

### CODE REVIEW

**Comment by friedmud** [comment]
A couple of things this still needs:

- [ ] Finish the test.  It's all there - just need to make a `Python` test and then put the execution of the script in the Python file.  I just don't have time tonight.

- [ ] A _bit_ of authentication.  Not much - probably just a `password` parameter that comes from the input file will be enough.

Note: this is _regular_ http - it is not secure.  It is fine if you're running both things on the same box... but if you are connecting remotely then you will need to open an ssh tunnel to tunnel the traffic through so it is not interceptable.

**Review by loganharbour — CHANGES_REQUESTED** [review]
New TestHarness tester?

**Comment by friedmud** [comment]
Yeah - is that thing @socratesgorilla did in yet?  That's what I need: just the ability to run a Python script as a test (and then do a CSV diff)

Testing this will also require that we have the Python "requests" package... which we should add as a dependency of our moose package in Conda.

**Comment by dschwen** [comment]
This adds a _second_ JSON library to MOSOE contrib?

**Comment by socratesgorilla** [comment]
> Yeah - is that thing @socratesgorilla did in yet? That's what I need: just the ability to run a Python script as a test (and then do a CSV diff)
> 
> Testing this will also require that we have the Python "requests" package... which we should add as a dependency of our moose package in Conda.

The custom script logic is in MOOSE, just add `custom_evaluation_script = "myscript.py"` to your tests file. myscript.py should contain a function `check_output(output)` that does the logic, `output` is the stdout log of the test

**Comment by friedmud** [comment]
> This adds a _second_ JSON library to MOOSE contrib?

Yeah - there was no way around that for now.  It's intertwined in that server code.  Just think of it as _part_ of the server code ;-).  If someone has some free time and wants to dig through it and replace it with our other json library that's fine... but I also think it's ok to have this truly tiny json library.

I think it's worth it for what we get.  But I'm open to suggestions of alternatives!

**Comment by cticenhour** [comment]
@friedmud `requests` added to `moose-test-tools` (and therefore `moose-tools`) recipe as of 9692e93

File: modules/doc/content/newsletter/2023/2023_02.md [hunk 1]
```diff
@@ -7,6 +7,14 @@ for a complete description of all MOOSE changes.
 
 ## MOOSE Improvements
 
+### WebServerControl
+
+A new [/Control.md] object called [/WebServerControl.md] was added to the framework that allows for external control of a MOOSE-based simulation.
```
Comment (cticenhour) [inline modules/doc/content/newsletter/2023/2023_02.md:12]:
```suggestion
A new [/Controls/index.md] object called [/WebServerControl.md] was added to the framework that allows for external control of a MOOSE-based simulation.
```

**Comment by friedmud** [comment]
Hmmm.  I don't quite no how to handle these test failures.  They are failing because the socket is already used.  I'm guessing that it's because multiple testers are running in parallel?  Any thoughts @milljm @loganharbour ?

**Comment by dschwen** [comment]
Don't all tests in the same directory already run in serial (`parallel_scheduling` in `Tester.py` defaults to `False`)

**Comment by dschwen** [comment]
This would be a lot cooler using websockets instead of busy waiting. Maybe sometime in the future ...

**Comment by friedmud** [comment]
@dschwen this webserver does already have websocket support.  I chose not to use it for now due to wanting this to be simple to start... and to have maximum compatibility with everything out there (some people are going to call it from Matlab - I don't know about Matlab websocket support - but I know it can use a REST API).

After it goes in... feel free to try out the websocket support... we accept pull requests!

**Pull request closed without merging**